p2panda-rs/src/materialiser/error.rs

```diff
@@ -0,0 +1,12 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use thiserror::Error;
+
+/// Error types for methods of `Materializer` struct.
+#[derive(Error, Debug)]
+#[allow(missing_copy_implementations)]
+pub enum MaterialisationError {
+        /// Handle errors from `EntrySignedError` struct.
+    #[error(transparent)]
+    EntrySignedError(#[from] crate::entry::EntrySignedError),
```

Comment:
Thats maybe more an implicit "error handling pattern" we're using in `p2panda-rs`. We organize errors per module, so the `materialiser` module has its own Error type, the `entry` module has its own Error type and so on .. To know support handling errors coming from the outside they need to be converted. This can be done with the help of the `thiserror` crate and some magic macros. I think this separation helps a lot since Errors in Rust are relatively crazy and this gives clarity where something came from and how the dependencies are.

Errors in Rust is quite a [specific thing](https://www.lpalmieri.com/posts/error-handling-rust/) if you're interested in learning more :crying_cat_face:

Reply:
Thanks!